src/UseCases/AddSubscription/SubscriptionRequest.php

```diff
@@ -91,4 +91,27 @@ public function setWikilogin( bool $wikilogin ) {
 		$this->wikilogin = $wikilogin;
 	}
 
+	/**
+	 * Set the wikilogin value from the first value that matches /^(1|0|yes|no)$/
+	 *
+	 * @param array $values
+	 */
+	public function setWikiloginFromValues( array $values ) {
+		$trueValues = ['yes', '1'];
+		$falseValues = ['no', '0'];
+		$matchingValues = array_intersect( $values, array_merge( $trueValues, $falseValues ) );
+		$wikilogin = in_array( array_shift( $matchingValues ), $trueValues );
+		$this->setWikilogin( $wikilogin );
+	}
+
+	public static function createFromArray( array $values ): SubscriptionRequest {
+		$instance = new self();
+		foreach ( $values as $key => $value ) {
+			$accessor = 'set' . ucfirst( $key );
+			if ( is_callable( [$instance, $accessor ] ) ) {
+				call_user_func( [$instance, $accessor ], $value );
```

Comment:
This pattern also prevents using composition without adding shortcut methods. Looks like `SubscriptionRequest` could contain a `PesonName` value object with first and last name and an `Address` value object with street address, postcode and city.